### ISSUE TITLE
Deleting collapsed subdirectory breaks tree-view

### ISSUE THREAD

**Issue by marcoms** [atom/tree-view#1005]
### Prerequisites

* [x] Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
* Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
* Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
* [x] Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
* [x] Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

If there is single subdirectory which causes the tree-view to collapse it into a single entry like `folder-a/folder-b`, if the user attempts to delete it by right clicking and deleting it, the subdirectory is deleted, but the entry is still listed as `folder-a/folder-b`, when only `folder-a` exists. Attempting to delete it again using the tree-view gives an error ("The following file couldn't be moved to trash").

### Steps to Reproduce

1. Launch Atom
1. Ensure `tree-view.squashDirectoryNames` is `true`
1. Create a directory with a single subdirectory
2. Attempt to delete the collapsed subdirectory through the tree-view UI

**Expected behavior:** The directory should no longer be collapsed and display only the first directory

**Actual behavior:** The subdirectory is still listed although it is deleted

**Reproduces how often:** 100%

### Versions

```
Atom    : 1.13.0-beta8
Electron: 1.3.13
Chrome  : 52.0.2743.82
Node    : 6.5.0
```

tree-view version 0.211.1

### Additional Information

Collapsing and expanding the parent of the directory refreshes the state and displays properly.

Related #702

**Comment by miguelcg** [comment]
Pull request #1008 opened for this issue, feel free to review it! :)

**Comment by 50Wliu** [comment]
Fixed by #1008.
